x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/custom_query.ts

```diff
@@ -2419,5 +2422,341 @@ export default ({ getService }: FtrProviderContext) => {
         expect(alertsAfterEnable.hits.hits.length > 0).toEqual(true);
       });
     });
+
+    describe('@skipInServerlessMKI manual rule run', () => {
+      const { indexListOfDocuments } = dataGeneratorFactory({
+        es,
+        index: 'ecs_compliant',
+        log,
+      });
+
+      beforeEach(async () => {
+        await stopAllManualRuns(supertest);
+        await esArchiver.load('x-pack/test/functional/es_archives/security_solution/ecs_compliant');
+      });
+
+      afterEach(async () => {
+        await stopAllManualRuns(supertest);
+        await esArchiver.unload(
+          'x-pack/test/functional/es_archives/security_solution/ecs_compliant'
+        );
+      });
+
+      /*
+       * This test is to verify that when a rule is run manually in the past, on time range, which is not covered by schedule exexution
+       * it generates an alert
+       * and when the rule is run again in the past, it does not generate a duplicate alert.
+       */
+      it('should run rule in the past and generate alert, without duplicates', async () => {
```

Comment:
You might be able to eliminate the comment here:

```suggestion
      it('alerts when run on a time range that the rule has not previously seen, and deduplicates if run there more than once', async () => {
```

Although I think that you could certainly split this into two tests:

* alerts when run on a time range that the rule has not previously seen
* deduplicates alerts if manually run twice on the same time range

Reply:
Do you think we should do this kind of splitting in FTR? Execution of those tests pretty expensive, so I kind of follow similar best practice from [cypress](https://docs.cypress.io/guides/references/best-practices#Creating-Tiny-Tests-With-A-Single-Assertion)

Reply:
FTR tests are somewhere in between cypress and a unit test, IMO, but I always prefer two simpler (and yes, slower) tests to a single complex one, but that's subjective.

Tests should focus on documentation and coverage; performance is a secondary concern, and when you start optimizing for it, those other attributes start to suffer.